misc-scripts/xref_mapping/XrefParser/MGI_Desc_Parser.pm

```diff
@@ -57,53 +56,41 @@ sub run {
   my $syn_count = 0;
   my %acc_to_xref;
 
-
-    my $header = $mgi_io->getline(); #discard header line
-
-    chomp($header);
-    # crude emergency check for potentially altered file format.
-    my $header_template = qq(MGI Accession ID\tChr\tcM Position\tgenome coordinate start\tgenome coordinate end\tstrand\tMarker Symbol\tStatus\tMarker Name\tMarker Type\tFeature Type\tMarker Synonyms (pipe-separated));
-    if ($header ne $header_template) {die "File header has altered from format parser expects. Check MGI "};
-      
-    while ( my $line = $mgi_io->getline() ) {
-
-        chomp($line);
-        my ($accession, $chromosome, $position, $start, $end, $strand,$label, 
-            $status, $marker, $marker_type, $feature_type, $synonym_field) = split(/\t/,$line);
-            
-        $position =~ s/^\s+// if ($position);
-
-        my @synonyms = split(/\|/,$synonym_field) if ($synonym_field);
-        
-	
-        my $desc;
-	if ($marker) {
-	    $desc = $marker;
-	}
-        
+  my $input_file = Text::CSV->new({
+                                    sep_char           => "\t",
+				    empty_is_undef     => 1,
+                                    strict => 1,
+                                    auto_diag => 1,
+                                    allow_loose_quotes => 1,
+				   }) or croak "Cannot use file $file: ".Text::CSV->error_diag ();
+
+  # skip header
+  $input_file->getline($mgi_io);
+  $input_file->column_names([qw(accession chromosome position start end strand label status marker marker_type feature_type synonym_field)] );
+  while ( my $data = $input_file->getline_hr( $mgi_io ) ) {
+        my $accession = $data->{'accession'};
+        my $marker = defined($data->{'marker'}) ? $data->{'marker'} : undef;
         $acc_to_xref{$accession} = $self->add_xref({ acc        => $accession,
-    	                           		             label      => $label,
-    					                             desc       => $desc,
-    					                             source_id  => $source_id,
-    					                             species_id => $species_id,
-                                                                     dbi        => $dbi,
-    					                             info_type  => "MISC"} );
-        if($verbose and !$desc){
+                                                     label      => $data->{'label'},
+                                                     desc       => $marker,
+                                                     source_id  => $source_id,
+                                                     species_id => $species_id,
+                                                     dbi        => $dbi,
+                                                     info_type  => "MISC"} );
+        if($verbose and !$marker){
```

Comment:
perlcritic says Mixed high and low-precedence booleans at line 80, column 12.  See page 70 of PBP.  (Severity: 4)
`&&` should be used instead of `and` here